### PR TITLE
Progress bars

### LINKED ISSUE #66: Add loading indicators
When lists are empty (queue and person) and we're loading we should display the correct loading indicator instead of empty list state.
http://www.google.com/design/spec/components/progress-activity.html#progress-activity-behavior

[Video Example](https://material-design.storage.googleapis.com/publish/v_2/material_ext_publish/0B0NGgBg38lWWYXJ3Wmk1czNwUzg/components-progressactivity-behavior-061102_Loading_Files_Linear_xhdpi_002.webm)

### PR DESCRIPTION
**Pull request by giladgo**
Fixes #66 
### THE SCREENSHOT ANIMATION HAS A 3-SECOND FAKE DELAY ADDED

progress

### CODE REVIEW

File: app/app.iml [hunk 1]
```diff
@@ -24,7 +25,7 @@
   </component>
   <component name="NewModuleRootManager" inherit-compiler-output="false">
     <output url="file://$MODULE_DIR$/build/intermediates/classes/debug" />
-    <output-test url="file://$MODULE_DIR$/build/intermediates/classes/test/debug" />
+    <output-test url="file://$MODULE_DIR$/build/intermediates/classes/androidTest/debug" />
```
Comment (yonbergman) [inline app/app.iml]: why?

File: app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java [hunk 2]
```diff
@@ -82,12 +88,40 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void refreshQueue(CurrentSessionStore.SessionUpdateEvent event) {
         mQueueItems = mCurrentSessionStore.getQueue();
         mQueueAdapter.notifyDataSetChanged();
+        Log.v(LOG_TAG, "On UI Thread? " + Boolean.toString(Looper.getMainLooper().getThread() == Thread.currentThread()));
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java]: remove

File: app/src/main/java/com/iic/mokojin/cloud/getters/GetSessionData.java [hunk 3]
```diff
@@ -23,24 +24,24 @@
     public static Task<Pair<Match, List<QueueItem>>> getSessionData() {
         return ParseCloud.<HashMap<String, Object>>callFunctionInBackground(GET_SESSION_DATA_CLOUD_FUNCTION_NAME, new HashMap<String, Object>())
                 .onSuccess(new Continuation<HashMap<String, Object>, Pair<Match, List<QueueItem>>>() {
-            @Override
```
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/cloud/getters/GetSessionData.java:26]: Changes here are just whitespace

File: app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java [hunk 4]
```diff
@@ -82,12 +88,40 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                Log.v(LOG_TAG, "Showing data");
+                mEmptyView.setVisibility(View.INVISIBLE);
+                mListViewContainer.setVisibility(View.VISIBLE);
+            } else {
+                Log.v(LOG_TAG, "Showing empty view");
+                mEmptyView.setVisibility(View.VISIBLE);
+                mListViewContainer.setVisibility(View.INVISIBLE);
+            }
+        } else {
+            Log.v(LOG_TAG, "Showing progress bar");
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java]: remove logs

File: app/src/main/res/layout/choose_character_fragment.xml [hunk 5]
```diff
@@ -1,13 +1,41 @@
...
+            android:textSize="20sp"
+            />
+
+        <ProgressBar
+            android:indeterminate="true"
+            style="@android:style/Widget.DeviceDefault.ProgressBar.Horizontal"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content" />
+
+    </LinearLayout>
```
Comment (yonbergman) [inline app/src/main/res/layout/choose_character_fragment.xml]: Can we somehow extract this to a view?

Comment (yonbergman) [inline app/src/main/res/layout/choose_character_fragment.xml]: And control the text ?

**Comment by yonbergman** [comment]
Almost

File: app/app.iml (hunk 1 above)
Comment (giladgo) [inline app/app.iml]: fixed :+1:

File: app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java (hunk 2 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java]: :+1:

File: app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java (hunk 4 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/activities/PlayerQueueFragment.java]: :+1:

**Comment by udiWertheimer** [comment]
what yon said

**Pull request merged**